src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -329,7 +329,19 @@ private static string GetVsRootFromMSBuildAssembly(string msBuildAssembly)
 
         private static string GetMSBuildExeFromVsRoot(string visualStudioRoot)
         {
-            return FileUtilities.CombinePaths(visualStudioRoot, "MSBuild", CurrentToolsVersion, "Bin", "MSBuild.exe");
+            return FileUtilities.CombinePaths(
+                visualStudioRoot,
+                "MSBuild",
+                CurrentToolsVersion,
+                "Bin",
+#if NET35
+                IntPtr.Size == 8
+#else
+                System.Runtime.InteropServices.Marshal.SizeOf<IntPtr>() == 8
+#endif
```

Comment:
We have this, but it's not easily accessible in this file and uses .NET 4+ features in both the interesting function and other functions in the file that make it very annoying to use.

https://github.com/microsoft/msbuild/blob/13cfe7fc5a97772e4045671c71fa31a3c0856af0/src/Shared/EnvironmentUtilities.cs#L11

So I duplicated the code and included a net35-compatible implementation.